### PR TITLE
feat: Minkowski penalties for Ellipse-Polygon

### PR DESCRIPTION
**Pull request by jiriminarcik**
# Description

This PR implements the simplest version of implicit-based Minkowski constraints for `Ellipse` and `PolygonLike` shapes. This case requires only quadratic roots, which can be computed analytically without the use of #906.

# Implementation strategy and design decisions

Simple interface for implicit shapes was added `ImplicitEllipse` and `ImplicitHalfPlane` to make the code more readable.

# Examples with steps to reproduce them

To test triangle-ellipse overlap, run `npx roger watch maze.sub ellipse-maze.sty maze.dsl` in `packages/examples/src/minkowski-tests/maze`.

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new ESLint warnings
- [x] I have reviewed any generated changes to the `diagrams/` folder

# Open questions

- The next step is to implement ellipse-ellipse constraints using #906.

### CODE REVIEW

File: packages/core/src/contrib/Minkowski.ts [hunk 1]
```diff
@@ -274,3 +277,171 @@ export const containsPolygonPoints = (
...
+): [ad.Num, ad.Num] => {
+  const c = div(lambda, mul(2, sub(lambda, 1)));
+  return [
+    sub(ei.x, mul(c, div(hpi.a, ei.a))),
+    sub(ei.y, mul(c, div(hpi.b, ei.b))),
+  ];
+};
+
+/**
+ * Helper for Signed Distance Function (SFD) of a polygon and ellipse.
```
Comment (samestep) [inline packages/core/src/contrib/Minkowski.ts:388]:
```suggestion
 * Helper for Signed Distance Function (SDF) of a polygon and ellipse.
```

File: packages/core/src/contrib/Minkowski.ts [hunk 2]
```diff
@@ -274,3 +277,171 @@ export const containsPolygonPoints = (
...
+ */
+export const overlappingPolygonPointsEllipse = (
+  polygonPoints: ad.Num[][],
+  ellipse: Ellipse,
+  padding: ad.Num
+): ad.Num => {
+  const center = ops.vdiv(polygonPoints.reduce(ops.vadd), polygonPoints.length);
+  // Create a list of all sides given by two subsequent vertices
+  const sides = Array.from(
+    { length: polygonPoints.length },
```
Comment (samestep) [inline packages/core/src/contrib/Minkowski.ts:437]: Nice, I didn't realize you could do this!

File: packages/core/src/contrib/Minkowski.ts [hunk 3]
```diff
@@ -274,3 +277,171 @@ export const containsPolygonPoints = (
...
+ */
+const pointCandidates = (
+  ei: ImplicitEllipse,
+  hpi: ImplicitHalfPlane,
+  lambda: ad.Num
+): [ad.Num, ad.Num] => {
+  const c = div(lambda, mul(2, sub(lambda, 1)));
+  return [
+    sub(ei.x, mul(c, div(hpi.a, ei.a))),
+    sub(ei.y, mul(c, div(hpi.b, ei.b))),
```
Comment (samestep) [inline packages/core/src/contrib/Minkowski.ts:383]: From the original document, like these should both be `add`:
```suggestion
    add(ei.x, mul(c, div(hpi.a, ei.a))),
    add(ei.y, mul(c, div(hpi.b, ei.b))),
```

**Review by samestep — APPROVED** [review]
Looks great modulo the changes we discussed!

File: packages/core/src/contrib/Minkowski.ts (hunk 3 above)
Comment (jiriminarcik) [reply packages/core/src/contrib/Minkowski.ts]: Fixed! I also added more tests to catch that.

**Pull request merged**